CONTRIBUTING.md

```diff
@@ -218,443 +205,83 @@ If the feature or fix does not directly impact users, consider using a different
-4. wandb writes files to the local file system. When we're testing we need to make sure each test is isolated.
-5. wandb reads configuration state from global directories such as `~/.netrc` and `~/.config/wandb/settings` we need to override these in tests.
-6. The library needs to support jupyter notebook environments as well.
-
-To make our lives easier we've created lots of tooling to help with the above challenges. Most of this tooling comes in the form of [Pytest Fixtures](https://docs.pytest.org/en/stable/fixture.html). There are detailed descriptions of our fixtures in the section below. What follows is a general overview of writing good tests for wandb.
-
-To test functionality in the user process the `wandb_init_run` is the simplest fixture to start with. This is like calling `wandb.init()` except we don't actually launch the wandb backend process and instead returned a mocked object you can make assertions with. For example:
-
-```python
-def test_basic_log(wandb_init_run):
-    wandb.log({"test": 1})
-    assert wandb.run._backend.history[0]["test"] == 1
-```
-
-One of the most powerful fixtures is `live_mock_server`. When running tests we start a Flask server that provides our graphql, filestream, and additional web service endpoints with sane defaults. This allows us to use wandb just like we would in the real world. It also means we can assert various requests were made. All server logic can be found in `tests/utils/mock_server.py` and it's really straight forward to add additional logic to this server. Here's a basic example of using the `live_mock_server`:
-
-```python
-def test_live_log(live_mock_server, test_settings):
-    run = wandb.init(settings=test_settings)
-    run.log({"test": 1})
-    ctx = live_mock_server.get_ctx()
-    first_stream_hist = utils.first_filestream(ctx)["files"]["wandb-history.jsonl"]
-    assert json.loads(first_stream_hist["content"][0])["test"] == 1
-```
-
-Notice we also used the `test_settings` fixture. This turns off console logging and ensures the run is automatically finished when the test finishes. Another really cool benefit of this fixture is it creates a run directory for the test at `tests/logs/NAME_OF_TEST`. This is super useful for debugging because the logs are stored there. In addition to getting the debug logs you can find the `live_mock_server` logs at `tests/logs/live_mock_server.log`.
-
-We also have pytest fixtures that are automatically used. These include `local_netrc` and `local_settings` this ensures we never read those settings files from your own environment.
-
-The final fixture worth noting is `notebook`. This actually runs a jupyter notebook kernel and allows you to execute specific cells within the notebook environment:
-
-```python
-def test_one_cell(notebook):
-    with notebook("one_cell.ipynb") as nb:
-        nb.execute_all()
-        output = nb.cell_output(0)
-        assert "lovely-dawn-32" in output[-1]["data"]["text/html"]
+pip install -r requirements_dev.txt
```

Comment:
Shouldn't `requirements_test.txt` cover all this?

Reply:
what do you mean?

Reply:
should we have contributors just do

```bash
pip install -r requirements_test.txt -r requirements_dev.txt
```

in one go? I think the text explaining it doesn't quite make sense. It says that `requirements_dev.txt` is needed for tests, but then it's not clear what is the difference from `requirements_test.txt`.